test/backup.py

```diff
@@ -218,19 +228,24 @@ def _restore(self, t, tablet_type='replica'):
     else:
       t.check_db_var('rpl_semi_sync_slave_enabled', 'OFF')
       t.check_db_status('rpl_semi_sync_slave_status', 'OFF')
-
-  def _reset_tablet_dir(self, t):
+    logging.debug("done restoring tablet %s",str(datetime.datetime.now()))
+    
+  def _reset_tablet_dir(self, t, teardown=True):
     """Stop mysql, delete everything including tablet dir, restart mysql."""
-    extra_args = ['-db-credentials-file', db_credentials_file]
-    utils.wait_procs([t.teardown_mysql(extra_args=extra_args)])
+
+    extra_args = ['-db-credentials-file', db_credentials_file]    
+    if teardown:
+      utils.wait_procs([t.teardown_mysql(extra_args=extra_args)])
```

Comment:
the teardown_mysql is conditioned on teardown, but the subsequent call to init_mysql is not.
Can that lead to errors if someone calls this with teardown=false?

Reply:
These tests have been refactored. There were errors here that needed to be addressed